viral/grosmark_analysis.py

```diff
@@ -45,12 +45,77 @@ def grosmark_place_field(
     spks: np.ndarray,
     rewarded: bool | None,
     config: GrosmarkConfig,
+    plot: bool = True,
 ) -> None:
-    """The position of the animal during online running epochs on the 2-m-long run belts was binned into 100,
-      2-cm spatial bins. For each cell, the as within spatial-bin firing rate was calculated across all bins
+    """
+    Grosmark et al. place field analysis.
+    1. get place cell mask
+    2. get peak indices and peak positions
+    3. do pair-wise correlations
+    """
+    if session.wheel_freeze is None:
+        spks = binarise_spikes(spks)
+    else:
+        # """Based on the observed differences in calcium activity waveforms between the online and
+        # offline epochs (Supplementary Fig. 2), a threshold of 1.5 m.a.d. was used for online running epochs,
+        # while a lower threshold of 1.25 m.a.d. were used for offline immobility epochs.""
+        online_spks = binarise_spikes(
+            spks[
+                :,
+                session.wheel_freeze.pre_training_end_frame : session.wheel_freeze.post_training_start_frame,
+            ],
+            mad_threshold=1.5,
+        )
+        offline_spks_pre, offline_spks_post = get_preactivation_reactivation(
+            flu=spks, wheel_freeze=session.wheel_freeze
+        )
+        # TODO: should pre and post be binarised as one?
+        offline_spks_pre = binarise_spikes(
+            offline_spks_pre,
+            mad_threshold=1.25,
+        )
+        offline_spks_post = binarise_spikes(offline_spks_post, mad_threshold=1.25)
+        spks = np.hstack([offline_spks_pre, online_spks, offline_spks_post])
```

Comment:
I would add an assertion to check this is the same size as the original spks matrix